model/app.model.js

```diff
@@ -7,15 +7,59 @@ const selectTopics = () => {
     });
 };
 
-const selectArticle = (id) => {
+const selectArticleById = (id) => {
     const query = "SELECT * FROM articles WHERE article_id = $1;";
     let params = [id];
     return db.query(query, params).then((result) => {
         return result.rows;
     });
 };
 
+const selectArticles = () => {
+    return db
+        .query(
+            `SELECT 
+            articles.article_id,
+            articles.author,
+            articles.title,
+            articles.topic,
+            articles.created_at,
+            articles.votes,
+            articles.article_img_url,
+            COALESCE(SUM(comments.comment_id),0) as comments_count
+    FROM articles LEFT JOIN comments
+    ON articles.article_id = comments.article_id
+    GROUP BY  articles.article_id, articles.author, articles.title, articles.topic, articles.created_at,   articles.votes, articles.article_img_url
+    ORDER BY articles.created_at DESC;`
+        )
+        .then((result) => {
+            return result.rows;
+        });
+};
+
+const selectAllCommentsForArticle = (id) => {
+    const query =
+        "SELECT * FROM comments WHERE article_id =$1 ORDER BY created_at ASC";
+    let params = [id];
+    return db.query(query, params).then((results) => {
```

Comment:
We need to add error handling here, as we could be returning an empty array, but we also need to determine WHY. This could be due to ...
article exists but has no comments.
OR
article does not exist.
You will need to check whether the article exists in order to determine this. If it does exist but has no comments, we want an empty array back. If it doesn't exist, we want an error